src/System.Private.CoreLib/shared/Internal/Win32/RegistryKey.cs

```diff
@@ -189,10 +189,8 @@ public unsafe string[] GetValueNames()
                             break;
                         case Interop.Errors.ERROR_MORE_DATA:
                             {
-                                char[] oldName = name;
-                                int oldLength = oldName.Length;
-                                name = null;
-                                ArrayPool<char>.Shared.Return(oldName);
+                                int oldLength = name.Length;
+                                ArrayPool<char>.Shared.Return(name);
```

Comment:
I delete the null assignment here and removed the ``` oldname``` array. @stephentoub can you take a look if this change is correct ?

Reply:
> if this change is correct?

There's a corner-case reliability concern with the change.  If you look down below, the finally block has the code `if (name != null) Return(name);`.  Now imagine that either `checked(oldLength * 2)` threw an overflow exception or Rent threw an OutOfMemoryException, or something along those lines.  We would have just returned name to the pool, then we throw an exception, and then the finally block sees that name isn't null and returns it to the pool again erroneously.  Now we have the same buffer in the pool twice, which is really bad.  So the code was null'ing out name before returning it to the pool, so that if the new rent operation throws, the buffer wouldn't be returned twice.

Now that I see the full code in question, I think you made the right decision earlier to make `name` be a `char[]?`; let's go back to that.  I was commenting on it because Rent will never return null, but I understand now you were doing that because of this null assignment.